kratos/solving_strategies/schemes/residual_based_bdf_custom_scheme.h

```diff
@@ -366,26 +366,45 @@ class ResidualBasedBDFCustomScheme
      * @brief Updating first time derivative (velocity)
      * @param itNode the node interator
      */
-
     inline void UpdateFirstDerivative(NodesArrayType::iterator itNode) override
     {
         // DOUBLES
         std::size_t counter = 0;
         for ( const auto& i_var : mDoubleVariable) {
-            double& dotun0 = itNode->FastGetSolutionStepValue(mFirtsDoubleDerivatives[counter]);
-            dotun0 = BDFBaseType::mBDF[0] * itNode->FastGetSolutionStepValue(i_var);
-            for (std::size_t i_order = 1; i_order < BDFBaseType::mOrder + 1; ++i_order)
-                dotun0 += BDFBaseType::mBDF[i_order] * itNode->FastGetSolutionStepValue(i_var, i_order);
+            if (!itNode->IsFixed(mFirtsDoubleDerivatives[counter])) {
+                double& dotun0 = itNode->FastGetSolutionStepValue(mFirtsDoubleDerivatives[counter]);
+                dotun0 = BDFBaseType::mBDF[0] * itNode->FastGetSolutionStepValue(i_var);
+                for (std::size_t i_order = 1; i_order < BDFBaseType::mOrder + 1; ++i_order)
+                    dotun0 += BDFBaseType::mBDF[i_order] * itNode->FastGetSolutionStepValue(i_var, i_order);
+            }
             counter++;
         }
 
         // ARRAYS
         counter = 0;
         for ( const auto& i_var : mArrayVariable) {
-            array_1d<double, 3>& dotun0 = itNode->FastGetSolutionStepValue(mFirtsArrayDerivatives[counter]);
-            noalias(dotun0) = BDFBaseType::mBDF[0] * itNode->FastGetSolutionStepValue(i_var);
+
+            const std::string& dvariable_name = (mFirtsArrayDerivatives[counter]).Name();
+            const VariableComponent<ComponentType>& dvar_x = KratosComponents< VariableComponent<ComponentType>>::Get(dvariable_name + "_X");
```

Comment:
you need to avoid this. You are manipulating a string on every node and then searching for the variable by this string.

Reply:
I know, but this is how is done right now, if I change taht right now you will complain that I am changing too many things

Reply:
jeje, my comments here would go more for the cosmetic changes in this PR rather than for this change.

speficifically, can't u do something like

~~~c++
std::vector< VarComponentType* > VarPointers done more carefully
VarPointers.push_back( &(KratosComponents< VarComponentType >( var_x_name ) ));
VarPointers.push_back( &(KratosComponents< VarComponentType >( var_y_name ) ));
VarPointers.push_back( &(KratosComponents< VarComponentType >( var_z_name ) ));
~~~

and then use it? 
I think that if you know that your variable is of component type than you should have no issue with this.

@pooyan-dadvand i CC you since  this is something we were discussing the other day, in a different context

Reply:
Better a map?, I would use a map to store the variables, or even better storing the key for a faster access. In any case thsi is not the scope of this PR

Reply:
KratosComponents is already a map  string vs variabletype

what is slow is exactly this access so what i am suggesting is a way to avoid that search, i mean do it a single time and not for every node

Reply:
In any case, who implemented this? it would be good to mention the author since what is done here should be avoided at all cost

Reply:
Me, more than year ago

Reply:
Thinking, i see what you suggest I already did in some processes in the Meshing application :+1: